### ISSUE TITLE
Documentation is unclear on whether or not regex filtering is supported while using the `MOMMY_FORBIDDEN_WORDS` setting

### ISSUE THREAD

**Issue by qeeg** [FWDekker/mommy#103]
I need this setting to support regex filtering with hex escapes so that I can fully avoid triggering my Tourette's when making use of this setting, as typing my triggers or even just reading them will trigger it, thus causing me to twitch VERY painfully. However, the documentation for this setting currently makes no mention of regex filtering at all.

**Comment by FWDekker** [comment]
you're right, regex filtering is not (yet?) supported. sorry if i was not clear and falsely got your hopes up. i assumed that exact matching would at least be somewhat useful in that regard.

i also assumed hex escapes would work inside the pattern matching, but unfortunately not. i will investigate how to make that work, it should be relatively easy for me to get working. (i'll let you know once that is implemented.)

if i manage to get both (1) hex escapes and (2) hex ranges (e.g. `\146-\148` for the range of lowercase letters `f` until `h`) working, how complete of a solution would that be for you?

**Comment by qeeg** [comment]
I would still need general regex functionality sadly, as many of my Tourette's triggers have a TON of spelling variations that ALL need to be avoided. Luckily, I already have a couple of regexes that cover ALL of them, or at least, should lol.

**Comment by FWDekker** [comment]
the easiest way for me to do this would be to filter out individual templates and variables _before_ they are assembled, but that might not be sufficient if you have complex multi-word triggers.

for example, let's say
* the templates are `%%name%% is going for a walk` and `i will visit %%name%%`,
* possible values for `%%name%%` are `john`, `jane`, and `jay`, and
* your triggers are `jon` and `john` (with regex `/joh?n/`).

with my approach, mommy will remove possible values for `%%name%%` that match the regex (i.e. remove `john`), so no triggering content will be displayed. mission accomplished :)

however, if the triggers are `visit jon` and `visit john` (with regex `/visit joh?n/`), then my approach won't work. mommy will check possible values for `%%name%%` with the regex, but since `john` doesn't match the regex, mommy will allow it, and potentially show triggering content :(

please let me know whether you think this solution will be good enough :)

**Comment by qeeg** [comment]
No, none of my triggers are multi-word, they're all very simple and EXTREMELY common single words. So that won't be necessary.

**Comment by FWDekker** [comment]
nice timing, i just created #109 :P

i will include the feature in v1.5.0. you can check which version you have with `mommy -v`. also, importantly, please note that, unfortunately, it _doesn't_ support hex escapes, only octal escapes. i wrote a bit more details about that in the pr, in #109.

**Comment by FWDekker** [comment]
mommy v1.5.0 has been released and is now available as explained in [the installation instructions](https://github.com/FWDekker/mommy#installation) :)
